dist/s_define.list

```diff
@@ -2,6 +2,8 @@
 API_CALL
 API_CALL_NOCONF
 API_SESSION_INIT
+DIAG_SINGLE_THREAD_CHECK_START
+DIAG_SINGLE_THREAD_CHECK_STOP
```

Comment:
This is outside of our documented namespace (so are the API_XXX and TXN_XXX macros, but we shouldn't make the problem worse).

Might be worth #undef of the API_XXX and TXN_XXX macros while you're there.

Reply:
I don't see any compelling reason this is outside the `WT_` namespace.  I suggest `WT_DIAG_THREAD_CHECK_*`

Reply:
I've created WT-3706 for the more general change here, and fixed my particular violation.